extensions/gitpod/src/remoteConnector.ts

```diff
@@ -379,63 +380,149 @@ export default class LocalApp extends Disposable {
 		}
 	}
 
+	private async getWorkspaceSSHDestination(workspaceId: string): Promise<string | undefined> {
+		const session = await vscode.authentication.getSession(
+			'gitpod',
+			['function:getWorkspace', 'function:getOwnerToken', 'function:getLoggedInUser', 'resource:default'],
+			{ createIfNone: true }
+		);
+
+		const gitpodHost = vscode.workspace.getConfiguration('gitpod').get<string>('host')!;
+		const serviceUrl = new URL(gitpodHost);
+
+		const workspaceInfo = await withServerApi(session.accessToken, serviceUrl.toString(), service => service.server.getWorkspace(workspaceId), this.logger);
+		const workspaceUrl = new URL(workspaceInfo.latestInstance!.ideUrl);
```

Comment:
I think we should handle gracefully missing latest instance, i.e. if someone used a URL for already stopped worksapce somehow